src/app/collective-rewards/rewards/components/ClaimRewardModal/ClaimRewardsModal.tsx

```diff
@@ -0,0 +1,86 @@
+import { FC, useState, useMemo } from 'react'
+import { useClaimBackerRewards } from '../../backers/hooks/useClaimBackerRewards'
+import { usePricesContext } from '@/shared/context'
+import { useHandleErrors } from '@/app/collective-rewards/utils'
+import { useBackerRewardsContext } from '@/app/collective-rewards/rewards/backers'
+import { getFiatAmount } from '@/app/collective-rewards/rewards/utils'
+import { ClaimRewardsModalView } from './ClaimRewardsModalView'
+import { TokenSymbol } from '@/components/TokenImage'
+import { getTokens } from '@/lib/tokens'
+import { ClaimRewardType } from './types'
+
+interface ClaimRewardsModalProps {
+  open: boolean
+  onClose: () => void
+}
+
+const tokens = getTokens()
+
+const getRewardTokenAddress = (value: ClaimRewardType) => {
+  switch (value) {
+    case 'rif':
+      return tokens.rif.address
+    case 'rbtc':
+      return tokens.rbtc.address
+    case 'all':
+    default:
+      return undefined // Claim all rewards
+  }
+}
+
+export const ClaimRewardsModal: FC<ClaimRewardsModalProps> = ({ open, onClose }) => {
+  const [selectedRewardType, setSelectedRewardType] = useState<ClaimRewardType>('all')
+
+  const { claimRewards, isClaimable, isLoadingReceipt, isPendingTx } = useClaimBackerRewards(
+    getRewardTokenAddress(selectedRewardType),
+  )
+
+  const { data, isLoading, error } = useBackerRewardsContext()
+  const { prices } = usePricesContext()
+
+  const rifPrice = prices[TokenSymbol.STRIF]?.price ?? 0
```

Comment:
I think that this should by dynamic, instead of having the tokens hardcoded we need to get from the utils.